tests/integration/tests/conftest.py

```diff
@@ -86,17 +86,73 @@ def pytest_configure(config):
         "no_setup: No setup steps (pushing snap, bootstrapping etc.) are performed on any node for this test.\n"
         "dualstack: Support dualstack on the instances.\n"
         "etcd_count: Mark a test to specify how many etcd instance nodes need to be created (None by default)\n"
-        "node_count: Mark a test to specify how many instance nodes need to be created\n",
+        "node_count: Mark a test to specify how many instance nodes need to be created\n"
+        "node_config: Mark a test to specify detail configuration for each created node\n",
     )
 
 
 @pytest.fixture(scope="function")
-def node_count(request) -> int:
+def node_config(request) -> List[dict]:
+    """Return a list of node configurations for the test.
+
+    The configuration can be specified using the `node_count` or `node_config` markers.
+    if `node_count` is used, the configuration for each node is empty, only the count is used.
+
+    example)
+    ```python
+    @pytest.mark.node_count(3)
+    def my_test(instances):
+        len(instances) == 3   # because of the node_count marker
+    ```
+
+    if `node_config` is used, the configuration returned is a list of dicts
+    where each dict represents the configuration for a node.
+
+    example)
+    ```python
+    @pytest.mark.node_config("snap", ["latest", "1.31"])
+    def my_other_test(instances):
+        len(instances) == 2   # because of the markers
+        # instances[0] has snap installed from latest track
+        # instances[1] has snap installed from 1.31 track
+    ```
+    or
+    ```python
+    @pytest.mark.node_config("snap,extra_args", [("latest", "something"), ("1.31", "something-else")])
```

Comment:
Hm, I think this can become really messy.

Maybe a config dict makes more sense? Then we could also move long configurations to a file